tests/TestCase.php

```diff
@@ -81,4 +83,21 @@ function (
             );
         }
     }
+
+    protected static function getFixerByName(string $name): AbstractFixer
+    {
+        static $fixers = null;
+
+        if (null === $fixers) {
+            $factory = new FixerFactory();
+            $factory->registerBuiltInFixers();
+
+            $fixers = [];
+            foreach ($factory->getFixers() as $fixer) {
+                $fixers[$fixer->getName()] = $fixer;
+            }
+        }
+
+        return $fixers[$name];
```

Comment:
I believe we need to check if fixer `$name` exists, and if not throw a meaningful exception, not a built-in notice for undefined key.

Reply:
I got this as a hint in PHPStorm, but what would Infection "say" about such never used check? :)

Reply:
Why unused? Similar check was previously in ,`RuleSetsTest`. It is pretty possible to hit this 🙂.

Reply:
> It is pretty possible to hit this 🙂.

How? Build-in fixers are always the same set.

Reply:
```php
function testOfSomeone() {
    ...::getFixerByName("foo_bar_baz_I_can_put_anything_here_and_typesystem_will_not_complain");
}
```

Reply:
> ```
> function testOfSomeone() {
>     ...::getFixerByName("foo_bar_baz_I_can_put_anything_here_and_typesystem_will_not_complain");
> }
> ```

Is this something you plan to add to tests soon?

I do not understand why you want to have check for something that will never happen 🤷🏼

Reply:
What Wirone brings (and I echo) is that method accept a string that you use as key for array, yet key can be any abstract string and not be part of that array. Thus we suggest to have explicit guard for it rather than expecting nobody to attempt to provide random string as argument.

Reply:
I am fully aware what the code is doing, what you are suggesting, and what non-existing situation you want to prevent.

I simply disagree with that approach - if someone would try to pass a random string then such test will fail, right?

Reply:
I guess we just don't enjoy random low-level warnings or violation of typehinting
ref https://3v4l.org/nNi4K3

Reply:
What @keradus is trying to say is exactly what I pointed out in my initial comment. It's not about the current code that is using only existing values, it's about controlling the code's behaviour for cases when someone does something wrong. @kubawerlos is right, test would fail, but with ugly warning/fatal error that just looks uncovered.

I believe we're wasting time and causing nerves, while it's clear that it's 3:1 here in favour of explicit check (me, Kerad and @julienfalque's 👍).